library/pkcs7.c

```diff
@@ -662,61 +715,30 @@ static int mbedtls_pkcs7_data_or_hash_verify(mbedtls_pkcs7 *pkcs7,
      * identification and SignerIdentifier fields first. That would also allow
      * us to distinguish between 'no signature for key' and 'signature for key
      * failed to validate'.
-     *
-     * We could also cache hashes by md, so if there are several sigs all using
```

Comment:
Last, bu a minor question,  seems the latest version of code assumes this case "if there are several sigs all using the same algo" is TRUE.  Is that what we want? or do I miss some description in spec RFC 2315?

Reply:
Seems I miss the comment` "/** For now, it assumes there is only one digest algorithm specified **/"` in function `pkcs7_get_digest_algorithm_set`(), it should be added into the TODO lists, I think.

Reply:
`pkcs7.h` documents that `Only one digest algorithm is supported in Mbed TLS.`. If multiple digest algorithms are given, we get a `MBEDTLS_ERR_PKCS7_FEATURE_UNAVAILABLE` error which is the right thing to do. So I think this is OK - it's clear that we don't support it, and we fail in the right way.

Reply:
(This is tested)